Refactor/organize filter and labels

Ryoken opened the pull request:
#### Any background context you want to provide?
see ticket

#### What's this PR do?
refactors the beta inventory list filters and settings

#### How should this be manually tested?
try filtering on the table and note the new filter in the upper right (see screenshots for examples)

#### What are the relevant tickets?
https://github.com/SEED-platform/seed/issues/3073

#### Screenshots (if appropriate)
If available, filtering by a label:
<img width="1027" alt="Screen Shot 2022-01-06 at 3 52 09 PM" src="https://user-images.githubusercontent.com/564231/148469858-cd548247-3b4e-4044-ab4f-e578c412b81c.png">

Table filters show up in the filter list:
<img width="1024" alt="Screen Shot 2022-01-06 at 3 51 46 PM" src="https://user-images.githubusercontent.com/564231/148469885-c4e691b1-8a01-448f-91eb-5cda45d772a6.png">

Splits multiple column filters:
<img width="700" alt="Screen Shot 2022-01-06 at 5 04 04 PM" src="https://user-images.githubusercontent.com/564231/148469992-7c2bcd4b-916a-4cce-989a-060334c9ddec.png">

Deleting only part of a filter updates the column filter:
<img width="699" alt="Screen Shot 2022-01-06 at 5 04 15 PM" src="https://user-images.githubusercontent.com/564231/148470021-0eca3063-6a92-4535-a243-b702ebe73579.png">

Actions have been organized and converted to a dropdown menu:
<img width="471" alt="Screen Shot 2022-01-06 at 5 09 34 PM" src="https://user-images.githubusercontent.com/564231/148470090-75e60ec9-20ad-459c-a800-f89efd92f1d8.png">

macintoshpie commented on seed/static/seed/js/controllers/inventory_list_beta_controller.js:
```diff
@@ -122,6 +122,20 @@ angular.module('BE.seed.controller.inventory_list_beta', [])
 
+      // remove editing on the filter list input (ngTagsInput doesn't support readonly yet)
+      let findFiltersListAttempts = 0;
+      const findFiltersList = setInterval(() => {
```
so this makes the filters input read only? not crazy about the setInterval, but looks like its our best option

macintoshpie commented on seed/static/seed/js/controllers/inventory_list_beta_controller.js:
```diff
@@ -1136,6 +1169,33 @@ angular.module('BE.seed.controller.inventory_list_beta', [])
       };
 
+      const operatorLookup = {
+        'ne': '!',
```
I think this might need to be `!=`, see video below -- `!=` end up turning into a `contains !<value>`

https://user-images.githubusercontent.com/18518728/148571455-5684d3df-cc1e-4889-896f-4a8b52e22f10.mov

Ryoken replied:
Ah yeah, good catch

macintoshpie commented on seed/static/seed/js/controllers/inventory_list_beta_controller.js:
```diff
@@ -1222,8 +1282,9 @@ angular.module('BE.seed.controller.inventory_list_beta', [])
-                  const {operator, value} = parseFilter(subFilter)
-                  $scope.column_filters.push({column_name, operator, value})
+                  const {string, operator, value} = parseFilter(subFilter)
+                  const display = [column_name, string, value].join(' ')
```
Do you think we could display the column's "display name" instead of the internal column_name here? Might be a better experience for users (e.g. showing `"Site EUI" < 10` instead of `site_eui < 10`)

Ryoken replied:
I had to collect all the display names on page load to support this but it's working.  It isn't too terribly helfpul for extra_data though!

macintoshpie reviewed (changes requested):
This is awesome! So nice to be able to see what's actually applied. A few comments:

1. Could we update the CSS s.t. it doesn't mess up the original inventory list view?
<img width="1368" alt="Screen Shot 2022-01-07 at 11 14 28 AM" src="https://user-images.githubusercontent.com/18518728/148572863-cd78a57d-3624-4939-95e7-64a8e943b740.png">

2. I do think it would be useful to see the currently applied sort at the top, and providing a shortcut to "remove" it

3. For the new "Current filters" box, could we:
  - change the mouse cursor so that it's not the "text" cursor when hovered over (since we can't actually type stuff into it)
  - make the box + trashcan appear disabled when there aren't any filters applied (like we do for the labels box)
  - add a placeholder to the box which says something like "Add filters by updating the filter above the column" or something like that so users know that they can't use that box itself to add filters?

Ryoken commented:
<img width="1029" alt="Screen Shot 2022-01-11 at 5 06 58 PM" src="https://user-images.githubusercontent.com/564231/149041217-64e18e68-cb94-422d-8c92-35fb88d5ff47.png">

Added sorting, removed all editable styles and added a placeholder.  The way the tag plugin works the placeholder is always displayed so I made it a shorter string.  If this looks good I'll update translations and merge!

RDmitchell commented:
@Ryoken / @macintoshpie / @axelstudios / @nllong -- this screen seems very busy. 

I think there might need to be a bit of UX design applied. 

Any reason we can't keep Cycle and Column List Profile where they currently are?

macintoshpie commented:
@RDmitchell we can definitely iterate on this deign, but I think we'd like to get this first stab at it merged in. Then we can deploy it for you and you can test it and give us your feedback.

macintoshpie reviewed (approved):
Great!

RDmitchell commented:
Sure, that's fine

On Wed, Jan 12, 2022, 7:42 AM Ted Summer ***@***.***> wrote:

> @RDmitchell <https://github.com/RDmitchell> we can definitely iterate on
> this deign, but I think we'd like to get this first stab at it merged in.
> Then we can deploy it for you and you can test it and give us your feedback.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SEED-platform/seed/pull/3075#issuecomment-1011177004>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABQFXRQWZAUBY57VWHCWQGDUVWOPXANCNFSM5LNRIDEA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
